src/stream/src/executor/hash_agg.rs

```diff
@@ -411,22 +433,30 @@ impl<K: HashKey, S: StateStore> HashAggExecutor<K, S> {
 
         let dirty_cnt = vars.group_change_set.len();
         if dirty_cnt > 0 {
-            // Produce the stream chunk
+            // Produce the stream chunk.
             let group_key_data_types =
                 &this.info.schema.data_types()[..this.group_key_indices.len()];
             let mut group_chunks =
                 IterChunks::chunks(vars.group_change_set.drain(), this.chunk_size);
             while let Some(batch) = group_chunks.next() {
-                let keys_in_batch = batch.into_iter().collect_vec();
+                let keys_in_batch: Vec<_> = batch
+                    .into_iter()
+                    .map(|hash_key| -> Result<_, ArrayError> {
+                        let row = hash_key.deserialize(group_key_data_types)?;
+                        Ok(TempGroupKey { hash_key, row })
+                    })
+                    .try_collect()?;
```

Comment:
> decrease caused by additional row construction

I guess it is mainly because of the `HashKey::deserialize` and I will do some benchmark and optimization on the HashKey implementation.

 https://github.com/risingwavelabs/risingwave/issues/8683

Reply:
I doubt whether it's because `HashKey::deserialize`, this deserialization existed before as `key.deserialize_to_builders`.🤔

Reply:
Probably worth taking a look at: 
1. Flamegraph for this.
2. Aggregation cached keys chart in grafana. See https://www.notion.so/risingwave-labs/Comparing-Nexmark-q17-modified-queries-ff916f8a25ef4956aed77886452b61a8#b28d9c5c9aad4da48e38f1cf1cd478f0 for instance.